metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/EnrichmentService.java

```diff
@@ -119,54 +68,16 @@ private void findEnrichmentEntitiesBySearchValue(
    * @param referenceValue The URI to check for match
    * @return the structured result of the enrichment
    */
-  public EnrichmentBase enrichByEquivalenceValues(ReferenceValue referenceValue) {
+  public List<EnrichmentBase> enrichByEquivalenceValues(ReferenceValue referenceValue) {
+    ReferenceTerm referenceTerm;
+    Map<ReferenceTerm, List<EnrichmentBase>> result = new HashMap<>();
     try {
-      final List<EntityType> entityTypes = referenceValue.getEntityTypes();
-      List<EnrichmentBase> foundEnrichmentBases = new ArrayList<>();
-      if (CollectionUtils.isEmpty(entityTypes)) {
-        foundEnrichmentBases = searchBasesFirstAboutThenOwlSameAs(referenceValue.getReference(),
-            null);
-      } else {
-        for (EntityType entityType : entityTypes) {
-          foundEnrichmentBases = searchBasesFirstAboutThenOwlSameAs(referenceValue.getReference(),
-              entityType);
-        }
-      }
-
-      if (CollectionUtils.isNotEmpty(foundEnrichmentBases)) {
-        return foundEnrichmentBases.get(0);
-      }
-    } catch (RuntimeException e) {
-      LOGGER.warn("Unable to retrieve entity from id", e);
+      referenceTerm = new ReferenceTermType(new URL(referenceValue.getReference()), referenceValue.getEntityTypes());
+      result = persistentEntityResolver.resolveByUri(Set.of(referenceTerm));
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
```

Comment:
And again... Perhaps instead of Strings you may wish to pass URL objects in certain places... although I'm not sure that helps here, actually. In this case, you may wish to throw a 400 bad request error.

Reply:
Fixed